src/webapi/xml_http_request.rs

```diff
@@ -0,0 +1,129 @@
+
+impl XMLHttpRequest {
+    /// Create new XMLHttpRequest
+    pub fn new() -> XMLHttpRequest {
+        XMLHttpRequest(js!{
+            return new XMLHttpRequest();
+        }.into_reference().unwrap())
+    }
+
+    /// Returns the current state of the request. 
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/readyState)
+    pub fn ready_state(&self) -> ReadyState {
+        use self::ReadyState::*;
+        let state: u16 = js!( return @{self}.readyState; ).try_into().unwrap();
+        match state {
+            0 => Unsent,
+            1 => Opened,
+            2 => HeadersReceived,
+            3 => Loading,
+            4 => Done,
+            _ => unreachable!( "Unexpected value of XMLHttpRequest::readyState: {}", state )
+        }
+    }
+
+    /// Returns a DOMString that contains the response to the request as text, or None
+    /// if the request was unsuccessful or has not yet been sent.
+    ///
+    ///[(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/responseText)
+    pub fn response_text(&self) -> Option<String> {
+        let response = js!(return @{self}.responseText;);
+        match response {
+            Value::Null => None,
+            Value::String(resp) => Some(resp),
+            _ => None,
+        }
+    }
+
+    /// Returns an unsigned short with the status of the response of the request.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/status)
+    pub fn status(&self) -> u16 {
+        js!(return @{self}.status;).try_into().unwrap()
+    }
+
+    /// Open connection with given method (ie GET or POST), and the url to hit
+    /// Async is set for now because it's more ergonimic in rust and it is good form in javascript
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/open)
+    pub fn open(&self, method: &str, url: &str) {
+        js! {
+            @{self}.open(@{method}, @{url}, true);
+        };
+    }
+
+    /// Send request on an open connection with no data
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/send)
+    pub fn send(&self) {
+        js! {
+            @{self}.send(null);
+        };
+    }
+
+    /// Send request on an open connection with string body
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/send)
+    pub fn send_with_string(&self, body: String) {
+        js! {
+            @{self}.send(@{body});
+        };
+    }
+
+    /// Send request on an open connection with a byte array body
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/send)
+    pub fn send_with_bytes(&self, body: &[u8]) {
+        js! {
+            @{self}.send(@{body});
+        };
+    }
+
+    /// The abort() method aborts the request if it has already been sent.
```

Comment:
You don't have to repeat the method name, e.g. `The abort() method` since it should be already obvious to the reader. (: